plugins/outputs/opentelemetry/opentelemetry.go

```diff
@@ -170,6 +199,20 @@ func (o *OpenTelemetry) Write(metrics []telegraf.Metric) error {
 	return err
 }
 
+// Sorted returns a copy of the metrics in time ascending order.  A copy is
+// made to avoid modifying the input metric slice since doing so is not
+// allowed.
+func sorted(metrics []telegraf.Metric) []telegraf.Metric {
+	batch := make([]telegraf.Metric, 0, len(metrics))
+	for i := len(metrics) - 1; i >= 0; i-- {
+		batch = append(batch, metrics[i])
+	}
+	sort.Slice(batch, func(i, j int) bool {
+		return batch[i].Time().Before(batch[j].Time())
+	})
+	return batch
+}
+
```

Comment:
Why do you need this copy? AFAICS you are creating _new_ batches when grouping the metrics into the timestamp-metrics map. You then also sort by timestamp so you don't need a sorting here, do you?